Calendar

MaksimusCRS opened the pull request:
- [DEMO LINK](https://MaksimusCRS.github.io/layout_calendar/)
- [TEST REPORT LINK](https://MaksimusCRS.github.io/layout_calendar/report/html_report/)

yura0seredyuk commented on src/index.html:
```diff
@@ -7,7 +7,6 @@
   </head>
   <body>
-    <h1>Calendar</h1>
     <div class="calendar calendar-30 calendar-tue">
```
не адаптивний календар , при зміні значень днів не сунеться контент , при зміні чисел в місяці вилазять лишні цифри

yura0seredyuk commented on src/styles/main.scss:
```diff
@@ -1,3 +1,47 @@
+}
+
+@for $i from 1 to 31 {
+  .calendar diV:nth-child(#{$i})::after {
```
Попробуй .calendar-28 і від 1 до 28 і так далі

yura0seredyuk commented on src/styles/main.scss:
```diff
@@ -1,3 +1,85 @@
+$days:
+  (
+    "sun": $max-margin,
+    "sut": $max-margin - ($cal-size + $cal-margin)*1,
```
"set"*

yura0seredyuk replied:
в рядку 73

solaryasha commented on src/styles/loops.scss:
```diff
@@ -0,0 +1,26 @@
+  @for $i from 1 to $counts+1 {
+    .calendar-#{$counts} div:nth-child(#{$i})::after {
+      content: "#{$i}";
+      text-align: center;
+      width: 100%;
+      font-size: 30px;
+      line-height: 100px;
```
https://mateacademy.slack.com/archives/G01HVTAUW92/p1611565588294100

solaryasha commented on src/styles/loops.scss:
```diff
@@ -0,0 +1,22 @@
+
+@for $i from 1 through 31 {
+  .calendar-#{$i} div:nth-child(-n+#{$i}) {
+    @include day;
```
Це правило повторюється 31 раз. Ти можеш задати його для `div` 1 раз і не використовувати міксін

nub1a replied:
Спасибо, не додумался)

solaryasha commented on src/styles/calendar.scss:
```diff
@@ -0,0 +1,42 @@
+      background-color: pink;
+    }
+  }
+  &-30 {
+    div:nth-last-child(1) {
+      display: none;
+    }
+  }
+  &-29 {
+    div:nth-last-child(-n+2) {
+      display: none;
+    }
+  }
+  &-28 {
+    div:nth-last-child(-n+3) {
+      display: none;
+    }
+  }
```
Можна і циклом такі правила згенерувати

solaryasha reviewed (approved):
Круто 🚀

MaksimusCRS committed:
Calendar

- [DEMO LINK](https://MaksimusCRS.github.io/layout_calendar/)
- [TEST REPORT LINK](https://MaksimusCRS.github.io/layout_calendar/report/html_report/)

Bohdan-Maksymiuk commented on src/styles/variables/variables.scss:
```diff
@@ -0,0 +1,15 @@
+$size-day: 98px;
```
можна вказувати box-sizing: border-box, щоб не гратися з розмірами border

Bohdan-Maksymiuk commented on src/styles/variables/variables.scss:
```diff
@@ -0,0 +1,15 @@
+$size-border: 1px;
+$size-gap: 1px;
+$global-size-day: $size-day + 2 * $size-border;
+$max-width-calendar: ($size-day + 2 * $size-border + 7 * $size-gap) * 7;
```
немає сенсу вказувати max-width, календар же ж по суті ми не будемо зменшувати (кожен день тижня має бути в своїй колонці), тому достатньо вказати просто width

Bohdan-Maksymiuk reviewed (changes requested):
передивися як в тебе рахує ширину календаря, бо вона більша ніж треба 
http://joxi.net/82Qyb7OHVYR0W2

MaksimusCRS committed:
Done

- [DEMO LINK](https://MaksimusCRS.github.io/layout_calendar/)
- [TEST REPORT LINK](https://MaksimusCRS.github.io/layout_calendar/report/html_report/)
